### PR TITLE
[Docs][474350] Document the new starter template for SSR

### PR DESCRIPTION
**Pull request by ancaemcken**
## Description / Motivation

Adds documentation for the new starter template for SSR using a Sitecore Experience Edge endpoint as requested in PBI 474350.

It includes a nav menu link under  `techniques/ssr/`.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update (non-breaking change; modified files are limited to the `/docs` directory)

### CODE REVIEW

File: docs/data/routes/docs/techniques/ssr/sitecore-experience-edge/en.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+title: SSR using Sitecore Experience Edge
+---
+
+# Server-side rendering using Sitecore Experience Edge
+
+You can configure server-side rendering (SSR) for your JSS application using a Sitecore Experience Edge endpoint. 
+
+Sitecore provides a [sample for server-side rendering using a Sitecore Experience Edge endpoint](https://github.com/Sitecore/jss/tree/dev/samples/node-headless-ssr-experience-edge), Node.js, and Express. It renders your JSS application, querying layout and dictionary data, without proxying the requests.
+
+> The sample for SSR using a Sitecore Experience Edge endpoint is experimental and not officially supported by Sitecore.
```
Comment (anastasiya29) [inline docs/data/routes/docs/techniques/ssr/sitecore-experience-edge/en.md:13]: I think there may be some other places where we list experimental samples; might be a good idea to search the docs directory for "experimental" to check.

File: docs/data/routes/docs/techniques/ssr/sitecore-experience-edge/en.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+2. A Sitecore API key. 
+
+3. A React, Vue or Angular application built with JSS that:
+
+   - Supports [server-side rendering in integrated mode](/docs/techniques/ssr/integrated-mode-ssr).
+
+   - Supports the Experience Edge GraphQL schema. 
+
+   > You will need to update your application's GraphQL components and queries to be compatible with the Experience Edge schema. See the [GraphQL-ConnectedDemo](https://github.com/Sitecore/jss/blob/master/samples/nextjs/src/components/graphql/GraphQL-ConnectedDemo.tsx) component and [the associated GraphQL query](https://github.com/Sitecore/jss/blob/dev/samples/nextjs/src/components/graphql/GraphQL-ConnectedDemo.graphql) in the Next.js sample application.
```
Comment (anastasiya29) [inline docs/data/routes/docs/techniques/ssr/sitecore-experience-edge/en.md:29]: @OlegJytnik - can you confirm that this statement is accurate? I thought that Headless Services GraphQL endpoint and Edge endpoint are supposed to function that same

File: docs/data/routes/docs/techniques/ssr/sitecore-experience-edge/en.md (hunk 1 above)
Comment (illiakovalenko) [reply docs/data/routes/docs/techniques/ssr/sitecore-experience-edge/en.md:13]: @anastasiya29 But `node-headless-proxy` sample is experimental too.

File: docs/data/routes/docs/techniques/ssr/sitecore-experience-edge/en.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+2. In your React/Vue/Angular app's root directory, build your JSS app for production with `jss build`.
+
+3. Copy the production build artifacts from your app (`/dist` or `/build` within the app) to the `sitecoreDistPath` set in your app's `package.json` under the SSR sample root path. For example `jss-edge/dist/<JSS app name>`.
+
+> Alternatively, change the `instancePath` in your JSS app's `scjssconfig.json` to the SSR sample root path, and then use `jss deploy files` to complete the deployment to the SSR sample.
+
+5. In `jss-edge/config.js`, set the values for:
+
+   - `appName` to the value set for you JSS application in `package.json`.
```
Comment (illiakovalenko) [inline docs/data/routes/docs/techniques/ssr/sitecore-experience-edge/en.md:51]: set for you -> set for your

**Review by illiakovalenko — CHANGES_REQUESTED** [review]

File: docs/data/routes/docs/techniques/ssr/sitecore-experience-edge/en.md (hunk 2 above)
Comment (ambrauer) [reply docs/data/routes/docs/techniques/ssr/sitecore-experience-edge/en.md:29]: This is speaking about the legacy/non-edge GraphQL schema which all of the other sample apps still use. Maybe we need to call that out specifically to be clear?

File: docs/data/routes/docs/techniques/ssr/sitecore-experience-edge/en.md (hunk 1 above)
Comment (ancaemcken) [reply docs/data/routes/docs/techniques/ssr/sitecore-experience-edge/en.md:13]: I haven't found a more "legalese" wording elsewhere. I emphasized "experimental" and I listed this sample in the Quick Start doc, under "Experimental Templates". If more need doing on that front, let me know.

**Review by ambrauer — APPROVED** [review]

**Review by anastasiya29 — APPROVED** [review]

**Pull request merged**